dotnet/src/Connectors/Connectors.Memory.Common/SqlFilterTranslator.cs

```diff
@@ -178,30 +181,33 @@ private void TranslateMember(MemberExpression memberExpression, bool isSearchCon
         switch (memberExpression)
         {
             case var _ when this.TryBindProperty(memberExpression, out var property):
-                this.GenerateColumn(property.StorageName, isSearchCondition);
+                this.GenerateColumn(property, isSearchCondition);
                 return;
 
             case var _ when TryGetCapturedValue(memberExpression, out var name, out var value):
-                this.TranslateCapturedVariable(name, value);
+                this.TranslateCapturedVariable(value);
                 return;
 
             default:
                 throw new NotSupportedException($"Member access for '{memberExpression.Member.Name}' is unsupported - only member access over the filter parameter are supported");
         }
     }
 
-    protected virtual void GenerateColumn(string column, bool isSearchCondition = false)
-        => this._sql.Append('"').Append(column.Replace("\"", "\"\"")).Append('"');
+    protected virtual void GenerateColumn(VectorStoreRecordPropertyModel column, bool isSearchCondition = false)
```

Comment:
I've changed the argument from `string` to `VectorStoreRecordPropertyModel` just to make sure all the derived types deal with `StorageName` in explicit way (they know this value is always escaped, while some random string may or may not be escaped).